File: 5:10/bufOverflow.c
/* from "Smashing the Stack for Fun and Profit"
   http://www.insecure.org/stf/smashstack.txt
*/

#include <string.h>

void mystrcpy(char* dest, const char* src) {
    int i = 0;
    while(src[i]) {
	dest[i] = src[i];
	i++;
    }
}

void func(char *str) {
    char buf[126];
    strcpy(buf, str);
}

int main() {
    char large[256];
    int i;

    for(i = 0; i < 256; i++)
	large[i] = 'A';
    func(large);

    return 0;
}

File: 5:10/BufOverflow.java
/* from "Smashing the Stack for Fun and Profit"
   http://www.insecure.org/stf/smashstack.txt
*/

class BufOverflow {
    static void func(char[] str) {
	char buf[] = new char[126];
	for (int i = 0; i < str.length; i++) {
	    buf[i] = str[i];
	}
    }

    public static void main(String[] args) {
	char large[] = new char[256];
	
	for(int i = 0; i < 256; i++)
	    large[i] = 'A';
	func(large);
    }
}
